### PR TITLE
Add timeout for port-forward test

### LINKED ISSUE #128742: Failure cluster [cefe0d5b...] `[sig-cli] Kubectl Port forwarding with a pod being removed should stop port-forwarding`
##### Error text:
```
[FAILED] Expected an error to have occurred.  Got:
    <nil>: nil
In [It] at: k8s.io/kubernetes/test/e2e/kubectl/portforward.go:613 @ 11/10/24 22:47:31.95

```
#### Recent failures:
[11/11/2024, 2:00:45 AM ci-containerd-e2e-ubuntu-gce](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-containerd-e2e-ubuntu-gce/1855868045222744064)
[11/11/2024, 12:35:57 AM ci-kubernetes-e2e-gci-gce-alpha-enabled-default](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-e2e-gci-gce-alpha-enabled-default/1855846905636982784)
[11/10/2024, 6:00:50 PM ci-containerd-e2e-ubuntu-gce](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-containerd-e2e-ubuntu-gce/1855747247937949696)
[11/10/2024, 5:25:57 PM ci-kubernetes-e2e-gci-gce-alpha-enabled-default](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-e2e-gci-gce-alpha-enabled-default/1855738691507458048)
[11/10/2024, 5:23:18 PM ci-cloud-provider-aws-e2e-kubetest2](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-cloud-provider-aws-e2e-kubetest2/1855737684882886656)

/kind failing-test

/sig cli

### PR DESCRIPTION
**Pull request by soltysh**
#### What type of PR is this?
/kind flake
/sig cli

#### What this PR does / why we need it:
After removing a pod in port-forward test we wait for an error from POST request. Since the POST doesn't have a timeout it hangs indefinitely, so instead we're hitting a DefaultPodDeletionTimeout. To make sure the POST fails this adds a timeout to ensure we'll always get that expected error, rather than nil.

#### Which issue(s) this PR fixes:
Fixes #128742

#### Special notes for your reviewer:
/assign @aojea @liggitt 

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/kubectl/portforward.go [hunk 1]
```diff
@@ -695,11 +695,11 @@ func wsWrite(conn *websocket.Conn, channel byte, data []byte) error {
 	return err
 }
 
-func post(url string, reader io.Reader, transport *http.Transport) (string, error) {
-	if transport == nil {
-		transport = utilnet.SetTransportDefaults(&http.Transport{})
+func post(url string, reader io.Reader, timeout time.Duration) (string, error) {
+	client := &http.Client{
+		Transport: utilnet.SetTransportDefaults(&http.Transport{}),
+		Timeout:   timeout,
```
Comment (liggitt) [inline test/e2e/kubectl/portforward.go:701]: Hmm… a client side timeout will keep the request from hanging, but is it actually testing what we want? I thought we wanted to make sure the portforward handler itself wouldn't hang

Comment (soltysh) [reply test/e2e/kubectl/portforward.go:701]: It's the current limitation of this and other tests relying on port-forward. I'm planning to rewrite it to have a control over the port-forward invocation, but that unfortunately requires a major re-write that I'm planning to handle in the next weeks.

Comment (liggitt) [reply test/e2e/kubectl/portforward.go:701]: if we actually hit this timeout, we'll get back a non-nil error, but that error will fail this check, right?

`gomega.Expect(err.Error()).To(gomega.Or(gomega.ContainSubstring("connection reset by peer"), gomega.ContainSubstring("EOF")))`

Comment (soltysh) [reply test/e2e/kubectl/portforward.go:701]: I need to run this check locally and try it out, I didn't have the time earlier today. I'll do that tomorrow first thing in the morning and will update accordingly.

Comment (soltysh) [reply test/e2e/kubectl/portforward.go:704]: Did some tests, and added there match for context timeout, which should happen in that case.

**Comment by liggitt** [comment]
/lgtm
/approve

I agree this will help the flake. I'd like to see us follow up to see if we can keep requests from hanging through portforward while a pod is tearing down so this client side timeout is not required

File: test/e2e/kubectl/portforward.go [hunk 2]
```diff
@@ -611,7 +611,10 @@ var _ = SIGDescribe("Kubectl Port forwarding", func() {
 
 			ginkgo.By("Check the client error")
 			gomega.Expect(err).To(gomega.HaveOccurred())
-			gomega.Expect(err.Error()).To(gomega.Or(gomega.ContainSubstring("connection reset by peer"), gomega.ContainSubstring("EOF")))
+			gomega.Expect(err.Error()).To(gomega.Or(
+				gomega.ContainSubstring("connection reset by peer"),
+				gomega.ContainSubstring("EOF"),
+				gomega.ContainSubstring("context deadline exceeded")))
```
Comment (liggitt) [inline test/e2e/kubectl/portforward.go:617]: there is apparently yet another possible error string we could encounter - https://github.com/kubernetes/kubernetes/issues/129527

**Pull request merged**